hail/python/hailtop/hailctl/batch/submit.py

```diff
@@ -0,0 +1,58 @@
+import asyncio
+
+import hailtop.batch as hb
+import hailtop.batch_client.client as bc
+from hailtop import pip_version
+from hailtop.aiotools.copy import copy_from_dict
+from hailtop.config import get_remote_tmpdir, get_user_config_path, get_deploy_config
+from hailtop.utils import unpack_comma_delimited_inputs
+
+def init_parser(parser):
+    parser.add_argument('name', type=str, help='Batch name')
+    parser.add_argument('script', type=str, help='Path to script')
+    parser.add_argument('--image-name', type=str, required=False,
+                        help='Name for Docker image. Defaults to hailgenetics/hail')
+    parser.add_argument('--files', nargs='+', action='append', default=[],
+                        help='Comma-separated list of files or directories to add to the working directory of job')
+
+
+async def async_main(args):
+    script = args.script
+    files = unpack_comma_delimited_inputs(args.files)
+    user_config = get_user_config_path()
+
+    remote_tmpdir = get_remote_tmpdir('hailctl batch submit')
+    def cloud_prefix(path):
+        return f'{remote_tmpdir}/{path}'
+
+    b = hb.Batch(name=args.name, backend=hb.ServiceBackend())
+    j = b.new_bash_job()
+    j.image(args.image_name or f'hailgenetics/hail:{pip_version()}')
+
+    local_files_to_cloud_files = [{'from': local, 'to': cloud_prefix(local)} for local in files]
+    await copy_from_dict(files=[
+        {'from': script, 'to': cloud_prefix(script)},
+        {'from': str(user_config), 'to': cloud_prefix(user_config)},
+        *local_files_to_cloud_files,
+    ])
+    for file in local_files_to_cloud_files:
+        local_file = file['from']
+        cloud_file = file['to']
+        in_file = b.read_input(cloud_file)
+        j.command(f'ln -s {in_file} {local_file}')
```

Comment:
You might need to defend against users mistakenly providing fully qualified paths to `--files`. I guess just os.basename is enough. Might want to verify the user didn't specify two files with the same name.

Reply:
Changed to use the relative path from the user's current directory, so as to be compatible with scripts reading from relative paths.